### PR TITLE
Add migration for Vertex AI models

### PR DESCRIPTION
**Pull request by brendanshean**
## WHAT
1. Add a new table `Evidence::AutomlModel`
2. Add some annotations to factory files
3. Fix linter error with factories

## WHY
1. This table will be used for Vertex AI automl models
2. They were missing
3. Inexplicably, this linter [error](https://app.circleci.com/pipelines/github/empirical-org/Empirical-Core/24482/workflows/b22b44f7-c132-4cf4-a58f-259eec60acff/jobs/286816?invite=true#step-102-11550_131) started showing up now

## HOW
1. An existing model Evidence::AutomlModel points to the `comprehension.automl_models`.  Create a new table `evidence.automl_models` that is very similar to `comprehension.automl_models` except `automl_model_id` becomes `model_external_id` and then add a new column `endpoint_external_id`.  The older platform allowed the prediction model to be called through the `model_external_id`, but on Vertex AI the `endpoint_external_id` is also required, so we persist that now. A subsequent PR will switch the what Evidence::AutomlModel tablename points to.
2. After running `rails db:migrate`, annotations appeared
3. Change constant name to string in relevant factories

### Screenshots
(If applicable. Also, please censor any sensitive data)

### Notion Card Links
(Please provide links to any relevant Notion card(s) relevant to this PR.)

PR Checklist | Your Answer
------------ | -------------
Have you added and/or updated tests? |  No.  Just migration files now.
Have you deployed to Staging? | YES
Self-Review: Have you done an initial self-review of the code below on Github? | YES
Spec Review: Have you reviewed the spec and ensured this meets requirements and/or matches design mockups? | N/A

### CODE REVIEW

**Review by happythenewsad — APPROVED** [review]
Are you planning to do a followon PR that drops `comprehension_auotml_models`?

**Review by happythenewsad — APPROVED** [review]

**Comment by brendanshean** [comment]
> Are you planning to do a followon PR that drops `comprehension_auotml_models`?

Thank you for the reminder.  I was planning on doing one in a month or so once we feel like the newer model is working properly.

File: services/QuillLMS/engines/evidence/Gemfile.lock [hunk 1]
```diff
@@ -1,7 +1,7 @@
 PATH
```
Comment (cissyxyu) [inline services/QuillLMS/engines/evidence/Gemfile.lock:1]: why is there a Gemfile.lock diff but no Gemfile changes?

**Review by cissyxyu — APPROVED** [review]

File: services/QuillLMS/engines/evidence/Gemfile.lock (hunk 1 above)
Comment (brendanshean) [reply services/QuillLMS/engines/evidence/Gemfile.lock:1]: Good question.  For some reason the version for evidence was out of sync with this [file](https://github.com/empirical-org/Empirical-Core/blob/513f60d4817e82fe7b0ef142551c4d06f5dc39ec/services/QuillLMS/engines/evidence/lib/evidence/version.rb#L6)

**Pull request merged**